kedro/runner/runner.py

```diff
@@ -112,6 +112,10 @@ def run(
             self._logger.info(
                 "Asynchronous mode is enabled for loading and saving data"
             )
+
```

Comment:
Can we add a comment about this? I also notice this line is duplicated:
`registered_ds = [ds for ds in pipeline.datasets() if ds in catalog]`

Maybe we can refactor this into one loop so we only loop `pipeline.datasets()` once?

Reply:
1. Kept one loop and added a comment about warm-up
2. The line is duplicated on purpose, there's a comment explaining the difference but I also renamed the variable to avoid confusion

Reply:
`registered_ds_no_runtime_patterns` is a bit of a mouthful, could we come up with a shorter name? E.g. `warm_concrete_ds` (not necessarily the best option, but just an example)?

Tbh, I don't mind the two loops - they show the purpose better without too much of a slowdown, but the current refactor is also ok.